specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,318 @@
+                    if (String.Equals(args.Uri, "calculator:///", StringComparison.OrdinalIgnoreCase))
+                    {
+                        // If this matches our desired protocol, then set the
+                        // event args to cancel the event and launch the
+                        // calculator app.
+                        ProcessStartInfo info = new ProcessStartInfo
+                        {
+                            FileName = args.Uri,
+                            UseShellExecute = true
+                        };
+                        Process.Start(info);
+                    } 
+                    else
+                    {
+                        // To display a custom dialog we cancel the launch, display
+                        // a custom dialog, and then manually launch the external protocol. 
+                        string text = "Launching External Protocol";
+                        if (args.InitiatingOrigin != "")
+                        {
+                            text += "from ";
+                            text += args.InitiatingOrigin;
+                        }
+                        text += " to ";
+                        text += args.Uri;
+                        text += "\n";
+                        text += "Do you want to grant permission?";
+                        string caption = "Launching External Protocol request";
+                        MessageBoxButton btnMessageBox = MessageBoxButton.YesNoCancel;
+                        MessageBoxImage icnMessageBox = MessageBoxImage.None;
+                        MessageBoxResult resultbox = MessageBox.Show(text, caption, btnMessageBox, icnMessageBox);
+                        switch (resultbox)
+                        {
+                            case MessageBoxResult.Yes:
+                                ProcessStartInfo info = new ProcessStartInfo
+                                {
+                                    FileName = args.Uri,
+                                    UseShellExecute = true
+                                };
+                                Process.Start(info);
+                                break;
+
+                            case MessageBoxResult.No:
+                                break;
+
+                            case MessageBoxResult.Cancel:
+                                break;
+                        }
+
+                    }
+                    
+                }
+            }, null);
+        }
+    };
+}
+
+```
+
+# API Notes
+
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+## Win32 C++
+    
+```IDL
+// This is the ICoreWebView2_16 interface.
+[uuid(cc39bea3-d6d8-471b-919f-da253e2fbf03), object, pointer_default(unique)] 
+interface ICoreWebView2_16 : ICoreWebView2_15 {
+  /// Add an event handler for the `LaunchingExternalProtocol` event.
+  /// The `LaunchingExternalProtocol` event is raised when a launch request is made to

```

Comment:
Is "launch request" a thing in WebView?

Reply:
Please use 'navigate' here instead since this only applies to navigating the document not fetch or img src=... and so on.